lib/improver/wind_direction.py

```diff
@@ -0,0 +1,404 @@
+        # equal operator.
+        round_angle = np.around(angle, 2)
+        angle = np.where(round_angle == 360, 0.0, angle)
+
+        return angle
+
+    @staticmethod
+    def find_r_values(complex_in):
+        """Find radius values from complex numbers.
+
+        Takes input wind direction in complex values and returns array
+        containing r values using Pythagoras theorem.
+
+        Args:
+            angle_deg (np.ndarray or float):
+                3D array or float - wind direction angles in complex numbers.
+
+        Returns:
+            (np.ndarray or float):
+                3D array or float - Radius values extracts from
+                complex numbers.
+        """
+
+        return np.sqrt(np.square(complex_in.real)+np.square(complex_in.imag))
+
+    @staticmethod
+    def calc_polar_mean_std_dev(wind_dir_complex, wind_dir_deg_mean,
+                                r_vals, r_thresh, realization_axis):
+        """Find standard deviation of polar numbers.
+
+        The average wind direction complex values represent the midpoint
+        between the different values and so have r values between 0-1.
+
+        1) From wind_dir_deg_mean - create a new set of complex values.
+           Therefore they will have the same angle but r is fixed as r=1.
+        2) Use fixed-r average complex values to find the distance between the
+           mean point and all the wind direction values
+        3) Find the average distance between the mean point and the wind
+           direction values. Large average distance == low confidence.
+        4) Normalise the average distance values to produce a confidence
+           value that is equivalent to the standard deviation
+
+        Args:
+            wind_dir_complex (np.ndarray):
+                3D array - wind direction angles in complex numbers.
+            wind_dir_deg_mean (np.ndarray):
+                3D array - average wind direction in angles.
+            r_vals (np.ndarray):
+                3D array - Radius taken from average complex wind direction
+                angle.
+            r_thresh (float):
+                Any r value below threshold is regarded as meaningless.
+            realization_axis (int):
+                Axis over which to average the arrays over.
+
+        Returns:
+            dist_from_mean_norm (np.ndarray):
+                3D array - The average distance from mean normalised - used
+                as a confidence value.
+        """
+
+        wind_dir_complex_mean_fixr = WindDirection.deg_to_complex(
+            wind_dir_deg_mean)
+
+        # Find difference in the distance between all the observed points and
+        # mean point with fixed r=1.
+        # For maths to work - the "wind_dir_complex_mean_fixr array" needs to
+        # be "tiled" so that it is the same dimension as "wind_dir_complex".
+        wind_dir_complex_mean_tile = np.tile(wind_dir_complex_mean_fixr,
```

Comment:
@carolinejjones Could you have a look at this in more detail?
At the moment we calculate the mean point. If we suppose that the angles are 45 and 315 degrees then the mean point is at 0 degrees with a radius = 0.707.
In this code - we recalculate the mean point so that it has the same angle but the radius is fixed at r=1.
I am not sure of the mathematical justification of this. Could you send me an email explaining this so that I can put it in the science presentation this Thursday?

Reply:
We are interested in the difference in angle of each of the ensemble members to the angle of the mean value so we need the mean point to be recalculated to have radius 1. Otherwise suppose we had angles 45.0, 315.0 an 0.0. The mean point is at 0 degrees with a radius of 0.804. The difference in angle between the member 0.0 and the mean point is obviously 0 but if we calculate the distance between the ensemble member of 0.0 as a complex number and the original mean point this would not be 0. Hence we need to recalculate the mean angle to have radius 1 to correctly compare.